### PR TITLE
CRv2: Use Sequel connection to find new and updated contacts

### PR DESCRIPTION
**Pull request by hacodeorg**
[PLC-893](https://codedotorg.atlassian.net/browse/PLC-893): Use Sequel to connection to find new and updated contacts to sync to Pardot.

Two queries in `ContactRollupsPardotMemory` to find new and updated contacts can take longer than 30s and cause ActiveRecord to time-out. 

They are slow because we join `contact_rollups_processed` to `contact_rollups_pardot_memory`, which are both 4M+ rows each.

The solution is to use Sequel connection instead of ActiveRecord, similar to what we did in https://github.com/code-dot-org/code-dot-org/pull/35119.

https://github.com/code-dot-org/code-dot-org/blob/f2a9032bfe7d13fb88021de0b512c81cb86dc3bf/dashboard/app/models/contact_rollups_pardot_memory.rb#L172-L189

https://github.com/code-dot-org/code-dot-org/blob/f2a9032bfe7d13fb88021de0b512c81cb86dc3bf/dashboard/app/models/contact_rollups_pardot_memory.rb#L191-L217

## Testing story
- Unit tests + local pipeline execution.
- Ran pipeline in an adhoc with connection to a production-clone db.
    ```ruby
    crv2 = ContactRollupsV2.new(is_dry_run: true); 
    crv2.build_and_sync; 
    crv2.report_results;
    ```
## Reviewer Checklist:
- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/lib/contact_rollups_v2.rb [hunk 1]
```diff
@@ -28,6 +31,19 @@ def self.execute_query_in_transaction(query)
...
 
+  # Execute a query in the dashboard database and returns query results.
+  # The query uses a Sequel or ActiveRecord connection depends on the current Rails environment.
+  def self.retrieve_query_results(query)
+    # @see comments in +execute_query_in_transaction+ method for explanation
+    # why we have to use ActiveRecord connection in a test environment.
+    if Rails.env.test?
+      ActiveRecord::Base.connection.exec_query(query)
+    else
+      # +Database#[]+ method actually returns a dataset and does not fetch records until needed.
```
Comment (bencodeorg) [inline dashboard/lib/contact_rollups_v2.rb]: Maybe clarify this comment with more detail or a link about what a "dataset" is?

File: dashboard/lib/contact_rollups_v2.rb [hunk 2]
```diff
@@ -9,6 +9,9 @@ class ContactRollupsV2
     query_timeout: MAX_EXECUTION_TIME_SEC
   )
 
+  # Execute a SQL query in a transaction in the dashboard database.
+  # Does not return query results.
```
Comment (bencodeorg) [inline dashboard/lib/contact_rollups_v2.rb:13]: Ok I think this seems clear now, but to confirm -- you would use `retrieve_query_results` if you need the results of a query, and `execute_query_in_transaction` if not?

Comment (hacodeorg) [reply dashboard/lib/contact_rollups_v2.rb:13]: That's correct. I can add more explanation to both methods.

File: dashboard/lib/contact_rollups_v2.rb (hunk 1 above)
Comment (hacodeorg) [reply dashboard/lib/contact_rollups_v2.rb]: I will clarify it as `Sequel::Dataset`

**Review by bencodeorg — APPROVED** [review]

**Pull request merged**